File: js-archive/binary-search.js
var search = function (nums, target) {
  let l = 0
  let r = nums.length - 1

  let ans = -1

  while (l <= r) {
    let mid = Math.floor((l + r) / 2)

    if (nums[mid] === target) {
      ans = mid
      break
    } else {
      if (nums[mid] > target) {
        r = mid - 1
      } else {
        l = mid + 1
      }
    }
  }

  return ans
};

console.log(search([-1, 0, 3, 5, 9, 12], 9))

File: go-archive/binary-search.go
func search(nums []int, target int) int {
    l:=0; r := len(nums) - 1

    for l <= r {
        m:=(l + r) / 2
        if (nums[m] == target) {return m}

        if (nums[m] > target) {
            r = m - 1
        } else {
            l = m + 1
        }
    }

    return -1
}